container-storage-setup.sh

```diff
@@ -240,6 +240,39 @@ get_config_options() {
   return 0
 }
 
+# Check if multiple overlay directories are supported and if overlay module
+# itself is supported on the system.
+can_mount_overlay() {
+  local dir="/run/container-storage-setup/"
+  local lower1="$dir/lower1"
+  local lower2="$dir/lower2"
+  local upper="$dir/upper"
+  local work="$dir/work"
+  local merged="$dir/merged"
+  local cmd
+
+  # Create multiple directories in /run
+  if ! mkdir -p $dir; then
+    Error "Failed to create directory $dir"
+    return 1
+  fi
+
+  cmd="mkdir -p $lower1 $lower2 $upper $work $merged"
+  if ! $cmd; then
+    Error "Failed to run $cmd"
+    return 1
+  fi
+
+  cmd="mount -t overlay -o lowerdir=$lower1:$lower2,upperdir=$upper,workdir=$work none $merged"
```

Comment:
Minor, but I wonder if we should create a new mount namespace for this to avoid touching the global one for our temporary tests?  A simple `unshare -m` prefix here should work and also avoids having to `umount` for cleanup.

Reply:
Sounds good. Will do. Hopefully "unshare" returns exit status of command properly.